packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,852 @@
+  switch (network) {
+    case Network.Mainnet:
+    case Network.Stagenet:
+      return NetworkId.Mainnet
+    case Network.Testnet:
+      return NetworkId.Stokenet
+  }
+}
+
+type PartialTransactionPreviewResponse = {
+  receipt: {
+    status: 'Succeeded' | 'Failed' | 'Rejected'
+    fee_summary: {
+      execution_cost_units_consumed: number
+      finalization_cost_units_consumed: number
+      xrd_total_execution_cost: string
+      xrd_total_finalization_cost: string
+      xrd_total_royalty_cost: string
+      xrd_total_storage_cost: string
+      xrd_total_tipping_cost: string
+    }
+  }
+}
+
+/**
+ * The main client for the Radix network which is then wrapped by the {Client} adapting it to have
+ * a {BaseXChainClient} interface.
+ */
+export class RadixSpecificClient {
+  private innerNetwork: number
+  private innerGatewayClient: GatewayApiClient
+
+  constructor(networkId: number) {
+    this.innerNetwork = networkId
+    this.innerGatewayClient = RadixSpecificClient.createGatewayClient(networkId)
+  }
+
+  // #region Getters & Setters
+  public set networkId(networkId: number) {
+    this.innerNetwork = networkId
+    this.innerGatewayClient = RadixSpecificClient.createGatewayClient(networkId)
+  }
+
+  public get networkId(): number {
+    return this.innerNetwork
+  }
+
+  public get gatewayClient(): GatewayApiClient {
+    return this.innerGatewayClient
+  }
+  // #endregion
+
+  // #region Public Methods
+  public async currentEpoch(): Promise<number> {
+    return this.innerGatewayClient.status.getCurrent().then((status) => status.ledger_state.epoch)
+  }
+
+  public async currentStateVersion(): Promise<number> {
+    return this.innerGatewayClient.status.getCurrent().then((status) => status.ledger_state.state_version)
+  }
+
+  public async fetchBalances(address: string): Promise<Balance[]> {
+    const fungibleResources = await this.fetchFungibleResources(address)
+    const nonFungibleResources = await this.fetchNonFungibleResources(address)
+    const fungibleBalances = this.convertResourcesToBalances(fungibleResources)
+    const nonFungibleBalances = this.convertResourcesToBalances(nonFungibleResources)
+    return fungibleBalances.concat(nonFungibleBalances)
+  }
+
+  private convertResourcesToBalances(
+    resources: FungibleResourcesCollectionItem[] | NonFungibleResourcesCollectionItem[],
+  ): Balance[] {
+    const balances: Balance[] = []
+    // Iterate through nonFungibleResources
+    const decimalDiff = XRD_DECIMAL - 8
+    resources.forEach((item) => {
+      if (item.aggregation_level === 'Global') {
+        const asset: Asset = {
+          chain: RadixChain,
+          symbol: item.resource_address,
+          ticker: item.resource_address,
+          synth: false,
+        }
+
+        // We need to do this because item.amount can be either string or number
+        // depending on the type of resource
+        let xrdAmount: number
```

Comment:
Following the previous comment. This could be simplied as

```ts
       const amount = typeof item.amount === 'string' ? parseFloat(item.amount) : item.amount

        const balance: Balance = {
          asset: asset,
          amount: assetToBase(assetAmount(amount, XRD_DECIMAL)),
        }
```

Reply:
makes sense to me